libs/common/NumUtils.cs

```diff
@@ -370,6 +484,38 @@ public static int NumDigitsInLong(long v, ref bool fNeg)
             return 19;
         }
 
+        /// <summary>
+        /// Return number of digits in given double number incluing the decimal part and `.` character
+        /// </summary>
+        /// <param name="v">Double value</param>
+        /// <returns>Number of digits in the integer part of the double value</returns>
+        public static int NumOfCharInDouble(double v, out int integerDigits, out byte signSize, out int fractionalDigits)
+        {
+            if (v == 0)
+            {
+                integerDigits = 1;
+                signSize = 0;
+                fractionalDigits = 0;
+                return 1;
+            }
+
+            Debug.Assert(!double.IsNaN(v) && !double.IsInfinity(v));
+
+            signSize = (byte)(v < 0 ? 1 : 0); // Add sign if the number is negative
+            v = Math.Abs(v);
+            integerDigits = (int)Math.Log10(v) + 1;
```

Comment:
Could you use our `NumDigitsInLong(long)` for this part? And Math.Floor for the long part?

Reply:
We can't, initially I did that but encountered issue in number bigger than log because double.MaxValue is very big than long.MaxValue.

double.MaxValue 1.7976931348623157E+308
long.MaxValue 9223372036854775807